src/metrics.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+var util = require('./util');
+var moment = require('moment');
+var eventObj = require('./core/event/event');
+
+class Metrics {
+    constructor(eventBus) {
+        var self = this;
+        self._eventBus = eventBus;
+
+        self._eventBus.addEventListener(eventObj.UNKNOWN_USER_INPUT, this, function(param) {
+            var emitParams = {
+                type: 'errorReport',
+                subtype: 'default',
+                doc: {
+                    input: param.input,
+                    date: param.date
+                },
+                _cb: param._cb
+            }
+            self._eventBus.emitEvent(eventObj.DATABASE_ADD, emitParams);
+        });
+
+        self._eventBus.addEventListener(eventObj.COMMAND_REPORT_ADD, this, function(param) {
+            var inputIndividualData = {
+                date: moment().format('x'),
+                input: param.inputParams
+            }
+            var emitParams = {
+                type: 'commandReport',
+                fetchParams: {
+                    'command': param.command
+                },
+                _cb: function(err, docs) {
+                    if (err) {
+                        console.log(err);
+                    }
+                    if (Object.keys(docs).length == 0) {
+                        var addEmitParams = {
+                            type: 'commandReport',
+                            subtype: 'default',
+                            doc: {
+                                command: param.command,
+                                totalUsage: 1,
+                                individualData: [inputIndividualData]
+                            },
+                            _cb: param._cb
+                        }
+                        self._eventBus.emitEvent(eventObj.DATABASE_ADD, addEmitParams);
+                    } else {
+                        var updateEmitParams = {
+                            type: 'commandReport',
+                            params: {
+                                'command': param.command
+                            },
+                            updates: {
+                                $push: {
```

Comment:
Hm this breaks the abstraction I set up with [CUP](https://docs.google.com/document/d/1Uuvep4IKdm09m6kj8AIHPOJKt41Ddfj3JyBvG2DhTvM/edit#heading=h.q5i0q5v6k664). Can you add push as one of the supported custom operations instead of using nedb api directly (also update the google doc)